File: sample/ruby/main_map_sample.rb
require 'voxelamming'
# require_relative 'voxelamming'
require_relative 'map_util'

room_name = "1000"
vox = Voxelamming::VoxelammingManager.new(room_name)
# vox = VoxelammingManager.new(room_name)

vox.set_box_size(0.1)
vox.set_build_interval(0.001)
vox.set_command('liteRender')

column_num, row_num = 257, 257
csv_file = '../map_file/map_38_138_100km.csv'
height_scale = 100
high_color = [0.5, 0, 0]
low_color = [0, 1, 0]
map_data = Voxelamming.get_map_data_from_csv(csv_file, height_scale)
boxes = map_data['boxes']
max_height = map_data['max_height']
# skip = 1  # high power device
skip = 2  # normal device
# skip = 4  # low power device

(row_num / skip).times do |j|
  (column_num / skip).times do |i|
#     puts "#{i} #{j}"
    x = i - column_num / (skip * 2).floor
    z = j - row_num / (skip * 2).floor
    y = boxes[j * skip][i * skip]

    if y >= 0
      r, g, b = Voxelamming.get_box_color(y, max_height, high_color, low_color)
#       print("r: #{r}, g: #{g}, b: #{b}\n")
      vox.create_box(x, y, z, r: r, g: g, b: b)
    end
  end
end

vox.send_data(name: 'map_sample')

File: sample/python/main_map_sample.py
# voxelammingパッケージからBuildBoxクラスをインポートします
from voxelamming import BuildBox
from map_util import get_map_data_from_csv, get_box_color

# 変数の設定
column_num, row_num = 257, 257
csv_file = '../map_file/map_38_138_100km.csv'
height_scale = 100
high_color = (0.5, 0, 0)
low_color = (0, 1, 0)

# Voxelammingアプリに表示されている部屋名を指定してください
room_name = "1000"
# BuildBoxクラスのインスタンスを生成します
build_box = BuildBox(room_name)

# ボクセルの設定を行います
build_box.set_box_size(1)
build_box.set_build_interval(0.001)
build_box.set_command('liteRender')  # 描画を軽くするためのコマンド

# ボクセルを配置するため、位置と色を設定します
map_data = get_map_data_from_csv(csv_file, height_scale)
boxes = map_data['boxes']
max_height = map_data['maxHeight']
# skip = 1  # high power device
skip = 2  # normal
# skip = 4  # low power device

for j in range(row_num // skip):
    for i in range(column_num // skip):
        print(i, j)
        x = i - column_num // (skip * 2)
        z = j - row_num // (skip * 2)
        y = boxes[j * skip][i * skip]
        r, g, b = get_box_color(y, max_height, high_color, low_color)

        if y > 0:
            build_box.create_box(x, y, z, r, g, b, 1)

# ボクセルデータをアプリに送信します。
build_box.send_data("main_map_sample")
